Project-Shared-Bookmarks/storage.js

```diff
@@ -0,0 +1,45 @@
+// This is a scaffolding file we have provided for you which allows you to manage stored data for your application.
+// It can be loaded into index.html.
+// You should not need to modify it to complete the project.
+
+/**
+ * Get a list of user ids
+ *
+ * @returns {string[]} List of user id strings
+ */
+export function getUserIds() {
+  return ["1", "2", "3", "4", "5"];
+}
+
+/**
+ * Get data associated with a specific user.
+ *
+ * @param {string} userId The user id to get data for
+ * @returns {any | null} The data associated with the user
+ */
+export function getData(userId) {
+  try {
+    return JSON.parse(localStorage.getItem(`stored-data-user-${userId}`));
+  } catch (_) {
+    return null;
+  }
```

Comment:
I worry the catch-all-return-null may make debugging harder - what do you think about:

```suggestion
  const value = localStorage.getItem(`stored-data-user-${userId}`);
  if (!value) {
    return value;
  }
  return JSON.parse(value);
```

(Same for the other one, too)

Reply:
My intention with the catch was mostly to avoid errors with parsing invalid JSON (i.e. they've stored a plain string or something).

I'm not worried about the no value case - because `getItem` returns `null` if there's no data for that key and `JSON.parse(null)` returns null (i.e. doesn't throw).

So I can see two options here: we keep it as-is (tradeoff: "papering over" a bad state not throwing an error) or drop the catch altogether (tradeoff: a possibly confusing error, but is more "realistic"/debuggable). I think I might be leaning towards the latter.

Reply:
Yeah I think I lean the latter, but I could easily be convinced either way.